### ISSUE TITLE
--useHttps = The system cannot find the file specified

### ISSUE THREAD

**Issue by iyerusad** [Azure/azure-functions-core-tools#1367]
Attempting to use https. http works fine.

Per https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#start
> --useHttps | Bind to https://localhost:{port} rather than to http://localhost:{port}. **By default, this option creates a trusted certificate on your computer.**

But when running from within Powershell terminal (within VS code) it fails:
```bat
PS> func host start --useHttps

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Azure Functions Core Tools (2.7.1158 Commit hash: f2d2a2816e038165826c7409c6d10c0527e8955b)
Function Runtime Version: 2.0.12438.0
Generating a self signed certificate using openssl
The system cannot find the file specified
```

Env Details
```
PS> $HOST
Name             : ConsoleHost
Version          : 5.1.17134.765
InstanceId       : e16fb76a-406b-4dc5-93ad-201eba939fad
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-US
CurrentUICulture : en-US
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace

PS> [environment]::OSVersion

Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.17134.0 Microsoft Windows NT 10.0.17134.0
```

Somewhat similar to #436,

**Comment by ahmelsayed** [comment]
what do you get if you do `where openssl`? do you have `openssl` cli installed on your machine? I think for some reason `where openssl` is successful, but then trying to call into `openssl` fails. 

Can you set `$Env:CLI_DEBUG=1` and run the same command again and share the output?

Also to unblock yourself, you can generate a cert using these commands


```powershell
$cert = New-SelfSignedCertificate -Subject localhost -DnsName localhost -FriendlyName "Functions Development" -KeyUsage DigitalSignature -TextExtension @("2.5.29.37={{text}}1.3.6.1.5.5.7.3.1")

# make sure to replace <password>
Export-PfxCertificate -Cert $cert -FilePath certificate.pfx -Password (ConvertTo-SecureString -String <password> -Force -AsPlainText)
```

then run using this. 

```
# replace <password>. it can either be a fileName that contains the password, or the actual password. 
func start --useHttps --cert certificate.pfx --password <password>
```

**Comment by ahmelsayed** [comment]
Also now dotnet has a `CertificateRequest` class for generating TLS certs. We should no longer need to depend on Powershell or Openssl cli for creating these certs. I'll use this issue to track removing that dependency.

**Comment by iyerusad** [comment]
Yea, I am familiar with the workaround, wanted to highlight documented scenario is not working.

No openssl in my path but some versions are shipped with various apps (e.g. git). Switching to dotnet class should serve best for different OSes. Looking forward to it.
```
ps> where openssl
ps>
ps> gci "C:\Program Files\Git\mingw64\bin\openssl.exe" | Select fullname

FullName
--------
C:\Program Files\Git\mingw64\bin\openssl.exe
```

**Comment by ahmedelnably** [comment]
@ahmelsayed can you please open a separate issue with the work needed.
Closing this issue as there is a mitigation
